lua: support static only builds

malbarbo opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Support building `pkgsStatic.lua`.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

malbarbo committed:
lua: support static only builds

symphorien commented on pkgs/development/interpreters/lua-5/interpreter.nix:
```diff
@@ -38,7 +39,9 @@ self = stdenv.mkDerivation rec {
   inherit patches;
 
-  postPatch = lib.optionalString (!stdenv.isDarwin) ''
+  inherit staticOnly;
```
is it really required ?

malbarbo replied:
No. I was a left over. Removed.

symphorien commented on pkgs/development/interpreters/lua-5/interpreter.nix:
```diff
@@ -56,7 +59,12 @@ self = stdenv.mkDerivation rec {
+    # building pkgsStatic.lua it fails because symbols from ncurses are not
+    # found. Adding ncurses here fixes the problem.
+    "MYLIBS=-lncurses"
+  ] else []);
```
is it possible to add it all time? This may prevent some regressions from going unnoticed

malbarbo replied:
This will require many rebuilds. I am trying to avoid that. Should I change it anyway?

symphorien replied:
as you can see with the ofborg badge, there are rebuilds anyway.

malbarbo replied:
Ok, I made the change.

symphorien commented on pkgs/development/interpreters/lua-5/interpreter.nix:
```diff
@@ -66,7 +74,8 @@ self = stdenv.mkDerivation rec {
     installFlagsArray=( TO_BIN="lua luac" INSTALL_DATA='cp -d' \
-      TO_LIB="${if stdenv.isDarwin then "liblua.${version}.dylib" else "liblua.a liblua.so liblua.so.${luaversion} liblua.so.${version}"}" )
+      TO_LIB="${if stdenv.isDarwin then "liblua.${version}.dylib"
+                else "liblua.a" + (if (!staticOnly) then " liblua.so liblua.so.${luaversion} liblua.so.${version}" else "")}" )
```
hum does the + apply to just liblua.a or the whole if then else ?

malbarbo replied:
Only to liblua.a. I added parentheses to make it clean.

SuperSandro2000 commented on pkgs/development/interpreters/lua-5/interpreter.nix:
```diff
@@ -56,7 +57,12 @@ self = stdenv.mkDerivation rec {
     "CC=${stdenv.cc.targetPrefix}cc"
     "RANLIB=${stdenv.cc.targetPrefix}ranlib"
-  ];
+  ] ++ (if staticOnly then [
+    # Lua links with readline wich depends on ncurses. For some reason when
+    # building pkgsStatic.lua it fails because symbols from ncurses are not
+    # found. Adding ncurses here fixes the problem.
+    "MYLIBS=-lncurses"
+  ] else []);
```
```suggestion
  ] ++ lib.optionals staticOnly [
    # Lua links with readline wich depends on ncurses. For some reason when
    # building pkgsStatic.lua it fails because symbols from ncurses are not
    # found. Adding ncurses here fixes the problem.
    "MYLIBS=-lncurses"
  ];
```

malbarbo replied:
:+1:

SuperSandro2000 commented on pkgs/development/interpreters/lua-5/interpreter.nix:
```diff
@@ -66,7 +72,8 @@ self = stdenv.mkDerivation rec {
 
     installFlagsArray=( TO_BIN="lua luac" INSTALL_DATA='cp -d' \
-      TO_LIB="${if stdenv.isDarwin then "liblua.${version}.dylib" else "liblua.a liblua.so liblua.so.${luaversion} liblua.so.${version}"}" )
+      TO_LIB="${if stdenv.isDarwin then "liblua.${version}.dylib"
+                else ("liblua.a" + (if (!staticOnly) then " liblua.so liblua.so.${luaversion} liblua.so.${version}" else ""))}" )
```
```suggestion
      TO_LIB="${if stdenv.isDarwin then "liblua.${version}.dylib"
                else ("liblua.a" + lib.optionalString (!staticOnly) " liblua.so liblua.so.${luaversion} liblua.so.${version}")}" )
```

malbarbo replied:
:+1:

r-rmcgibbo commented:
Result of `nixpkgs-review pr 127425` at 5674cff1 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages marked as broken and skipped:</summary>
  <ul>
    <li>clickshare-csc1</li>
    <li>googleearth-pro</li>
    <li>grass</li>
    <li>scribus</li>
    <li>vdrPlugins.xineliboutput</li>
  </ul>
</details>
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/359eecaf2ae03b2fad183e6ca53a7f8f">lua51Packages.digestif</a></li>
    <li><a href="https://gist.github.com/8251d88d2279c6b28e4910cd8c330c85">luaPackages.digestif</a> (lua52Packages.digestif)</li>
    <li><a href="https://gist.github.com/c8d9341d367b9da067f869e586f9fc8c">luaPackages.ltermbox</a> (lua52Packages.ltermbox)</li>
    <li><a href="https://gist.github.com/a6b5000322f4fc3cb1a04b7f5fb607cc">lua53Packages.ljsyscall</a></li>
    <li><a href="https://gist.github.com/ba3274eb1e661f7a09b44de7c5946ecd">lua53Packages.ltermbox</a></li>
    <li><a href="https://gist.github.com/eadcce9f00b7fa073137e916c8ea3a44">lua53Packages.luabitop</a></li>
    <li><a href="https://gist.github.com/7a9842d3471b43f5342167ed57d1c87a">lua53Packages.vstruct</a></li>
    <li><a href="https://gist.github.com/1d5ba581204a991307287da7bb01bca8">maxscale</a></li>
  </ul>
</details>
<details>
  <summary>584 packages skipped due to time constraints:</summary>
  <ul>
    <li>EBTKS</li>
    <li>adapta-gtk-theme</li>
    <li>aegisub</li>
    <li>alephone</li>
    <li>alephone-durandal</li>
    <li>alephone-eternal</li>
    <li>alephone-evil</li>
    <li>alephone-infinity</li>
    <li>alephone-marathon</li>
    <li>alephone-pathways-into-darkness</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>315 packages built successfully:</summary>
  <ul>
    <li>_90secondportraits</li>
    <li>alpine-make-vm-image</li>
    <li>apk-tools</li>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>bitsnbots</li>
    <li>bozohttpd</li>
    <li>dell-530cdn</li>
    <li>devilspie2</li>
    <li>dtrx</li>
    <li>fcitx5-lua</li>
    <li>fcitx5-with-addons</li>
    <li>grilo-plugins (gnome.grilo-plugins)</li>
    <li>gutenprintBin</li>
    <li>haskellPackages.hslua</li>
    <li>haskellPackages.hslua-module-path</li>
    <li>haskellPackages.hslua-module-system</li>
    <li>haskellPackages.hslua-module-text</li>
    <li>haskellPackages.tasty-lua</li>
    <li>highlight</li>
    <li>imapfilter</li>
    <li>instead</li>
    <li>kakounePlugins.quickscope-kak</li>
    <li>libsolv</li>
    <li>lite</li>
    <li>lsyncd</li>
    <li>lua (lua5 ,lua5_2_compat)</li>
    <li>lua51Packages.alt-getopt</li>
    <li>lua51Packages.ansicolors</li>
    <li>lua51Packages.argparse</li>
    <li>lua51Packages.basexx</li>
    <li>lua51Packages.binaryheap</li>
    <li>lua51Packages.bit32</li>
    <li>lua51Packages.busted</li>
    <li>lua51Packages.cassowary</li>
    <li>lua51Packages.cjson</li>
    <li>lua51Packages.compat53</li>
    <li>lua51Packages.cosmo</li>
    <li>lua51Packages.coxpcall</li>
    <li>lua51Packages.cyrussasl</li>
    <li>lua51Packages.dkjson</li>
    <li>lua51Packages.fifo</li>
    <li>lua51Packages.inspect</li>
    <li>lua51Packages.ldbus</li>
    <li>lua51Packages.ldoc</li>
    <li>lua51Packages.lgi</li>
    <li>lua51Packages.linenoise</li>
    <li>lua51Packages.ljsyscall</li>
    <li>lua51Packages.lpeg</li>
    <li>lua51Packages.lpeg_patterns</li>
    <li>lua51Packages.lpeglabel</li>
    <li>lua51Packages.lpty</li>
    <li>lua51Packages.lrexlib-gnu</li>
    <li>lua51Packages.lrexlib-pcre</li>
    <li>lua51Packages.lrexlib-posix</li>
    <li>lua51Packages.ltermbox</li>
    <li>lua5_1 (lua51Packages.lua)</li>
    <li>lua51Packages.lua-cmsgpack</li>
    <li>lua51Packages.lua-iconv</li>
    <li>lua51Packages.lua-lsp</li>
    <li>lua51Packages.lua-messagepack</li>
    <li>lua51Packages.lua-resty-http</li>
    <li>lua51Packages.lua-resty-jwt</li>
    <li>lua51Packages.lua-resty-openidc</li>
    <li>lua51Packages.lua-resty-openssl</li>
    <li>lua51Packages.lua-resty-session</li>
    <li>lua51Packages.lua-term</li>
    <li>lua51Packages.lua-toml</li>
    <li>lua51Packages.lua-yajl</li>
    <li>lua51Packages.lua-zlib</li>
    <li>lua51Packages.lua_cliargs</li>
    <li>lua51Packages.luabitop</li>
    <li>lua51Packages.luacheck</li>
    <li>lua51Packages.luacov</li>
    <li>lua51Packages.luadbi</li>
    <li>lua51Packages.luadbi-mysql</li>
    <li>lua51Packages.luadbi-postgresql</li>
    <li>lua51Packages.luadbi-sqlite3</li>
    <li>lua51Packages.luadoc</li>
    <li>lua51Packages.luaepnf</li>
    <li>lua51Packages.luaevent</li>
    <li>lua51Packages.luaexpat</li>
    <li>lua51Packages.luaffi</li>
    <li>lua51Packages.luafilesystem</li>
    <li>lua51Packages.lualogging</li>
    <li>lua51Packages.luaossl</li>
    <li>lua51Packages.luaposix</li>
    <li>lua51Packages.luarepl</li>
    <li>lua51Packages.luarocks</li>
    <li>lua51Packages.luarocks-nix</li>
    <li>lua51Packages.luasec</li>
    <li>lua51Packages.luasocket</li>
    <li>lua51Packages.luasql-sqlite3</li>
    <li>lua51Packages.luassert</li>
    <li>lua51Packages.luasystem</li>
    <li>lua51Packages.luautf8</li>
    <li>lua51Packages.luazip</li>
    <li>lua51Packages.luuid</li>
    <li>lua51Packages.luv</li>
    <li>lua51Packages.luxio</li>
    <li>lua51Packages.lyaml</li>
    <li>lua51Packages.markdown</li>
    <li>lua51Packages.mediator_lua</li>
    <li>lua51Packages.moonscript</li>
    <li>lua51Packages.mpack</li>
    <li>lua51Packages.nvim-client</li>
    <li>lua51Packages.penlight</li>
    <li>lua51Packages.readline</li>
    <li>lua51Packages.say</li>
    <li>lua51Packages.std__debug</li>
    <li>lua51Packages.std_normalize</li>
    <li>lua51Packages.stdlib</li>
    <li>lua51Packages.vicious</li>
    <li>lua51Packages.vstruct</li>
    <li>lua51Packages.wrapLua</li>
    <li>luaPackages.alt-getopt (lua52Packages.alt-getopt)</li>
    <li>luaPackages.ansicolors (lua52Packages.ansicolors)</li>
    <li>luaPackages.argparse (lua52Packages.argparse)</li>
    <li>luaPackages.basexx (lua52Packages.basexx)</li>
    <li>luaPackages.binaryheap (lua52Packages.binaryheap)</li>
    <li>luaPackages.bit32 (lua52Packages.bit32)</li>
    <li>luaPackages.busted (lua52Packages.busted)</li>
    <li>luaPackages.cassowary (lua52Packages.cassowary)</li>
    <li>luaPackages.cjson (lua52Packages.cjson)</li>
    <li>luaPackages.compat53 (lua52Packages.compat53)</li>
    <li>luaPackages.cosmo (lua52Packages.cosmo)</li>
    <li>luaPackages.coxpcall (lua52Packages.coxpcall)</li>
    <li>luaPackages.cqueues (lua52Packages.cqueues)</li>
    <li>luaPackages.cyrussasl (lua52Packages.cyrussasl)</li>
    <li>luaPackages.dkjson (lua52Packages.dkjson)</li>
    <li>luaPackages.fifo (lua52Packages.fifo)</li>
    <li>luaPackages.http (lua52Packages.http)</li>
    <li>luaPackages.inspect (lua52Packages.inspect)</li>
    <li>luaPackages.ldbus (lua52Packages.ldbus)</li>
    <li>luaPackages.ldoc (lua52Packages.ldoc)</li>
    <li>luaPackages.lgi (lua52Packages.lgi)</li>
    <li>luaPackages.linenoise (lua52Packages.linenoise)</li>
    <li>luaPackages.ljsyscall (lua52Packages.ljsyscall)</li>
    <li>luaPackages.lpeg (lua52Packages.lpeg)</li>
    <li>luaPackages.lpeg_patterns (lua52Packages.lpeg_patterns)</li>
    <li>luaPackages.lpeglabel (lua52Packages.lpeglabel)</li>
    <li>luaPackages.lpty (lua52Packages.lpty)</li>
    <li>luaPackages.lrexlib-gnu (lua52Packages.lrexlib-gnu)</li>
    <li>luaPackages.lrexlib-pcre (lua52Packages.lrexlib-pcre)</li>
    <li>luaPackages.lrexlib-posix (lua52Packages.lrexlib-posix)</li>
    <li>lua5_2 (lua52Packages.lua ,luaPackages.lua)</li>
    <li>luaPackages.lua-cmsgpack (lua52Packages.lua-cmsgpack)</li>
    <li>luaPackages.lua-iconv (lua52Packages.lua-iconv)</li>
    <li>luaPackages.lua-lsp (lua52Packages.lua-lsp)</li>
    <li>luaPackages.lua-messagepack (lua52Packages.lua-messagepack)</li>
    <li>luaPackages.lua-resty-http (lua52Packages.lua-resty-http)</li>
    <li>luaPackages.lua-resty-jwt (lua52Packages.lua-resty-jwt)</li>
    <li>luaPackages.lua-resty-openidc (lua52Packages.lua-resty-openidc)</li>
    <li>luaPackages.lua-resty-openssl (lua52Packages.lua-resty-openssl)</li>
    <li>luaPackages.lua-resty-session (lua52Packages.lua-resty-session)</li>
    <li>luaPackages.lua-term (lua52Packages.lua-term)</li>
    <li>luaPackages.lua-toml (lua52Packages.lua-toml)</li>
    <li>luaPackages.lua-yajl (lua52Packages.lua-yajl)</li>
    <li>luaPackages.lua-zlib (lua52Packages.lua-zlib)</li>
    <li>luaPackages.lua_cliargs (lua52Packages.lua_cliargs)</li>
    <li>luaPackages.luabitop (lua52Packages.luabitop)</li>
    <li>luaPackages.luacheck (lua52Packages.luacheck)</li>
    <li>luaPackages.luacov (lua52Packages.luacov)</li>
    <li>luaPackages.luadbi (lua52Packages.luadbi)</li>
    <li>luaPackages.luadbi-mysql (lua52Packages.luadbi-mysql)</li>
    <li>luaPackages.luadbi-postgresql (lua52Packages.luadbi-postgresql)</li>
    <li>luaPackages.luadbi-sqlite3 (lua52Packages.luadbi-sqlite3)</li>
    <li>luaPackages.luadoc (lua52Packages.luadoc)</li>
    <li>luaPackages.luaepnf (lua52Packages.luaepnf)</li>
    <li>luaPackages.luaevent (lua52Packages.luaevent)</li>
    <li>luaPackages.luaexpat (lua52Packages.luaexpat)</li>
    <li>luaPackages.luaffi (lua52Packages.luaffi)</li>
    <li>luaPackages.luafilesystem (lua52Packages.luafilesystem)</li>
    <li>luaPackages.lualogging (lua52Packages.lualogging)</li>
    <li>luaPackages.luaossl (lua52Packages.luaossl)</li>
    <li>luaPackages.luaposix (lua52Packages.luaposix)</li>
    <li>luaPackages.luarepl (lua52Packages.luarepl)</li>
    <li>luarocks (luaPackages.luarocks)</li>
    <li>luarocks-nix (luaPackages.luarocks-nix)</li>
    <li>luaPackages.luasec (lua52Packages.luasec)</li>
    <li>luaPackages.luasocket (lua52Packages.luasocket)</li>
    <li>luaPackages.luasql-sqlite3 (lua52Packages.luasql-sqlite3)</li>
    <li>luaPackages.luassert (lua52Packages.luassert)</li>
    <li>luaPackages.luasystem (lua52Packages.luasystem)</li>
    <li>luaPackages.luautf8 (lua52Packages.luautf8)</li>
    <li>luaPackages.luazip (lua52Packages.luazip)</li>
    <li>luaPackages.luuid (lua52Packages.luuid)</li>
    <li>luaPackages.luv (lua52Packages.luv)</li>
    <li>luaPackages.luxio (lua52Packages.luxio)</li>
    <li>luaPackages.lyaml (lua52Packages.lyaml)</li>
    <li>luaPackages.markdown (lua52Packages.markdown)</li>
    <li>luaPackages.mediator_lua (lua52Packages.mediator_lua)</li>
    <li>luaPackages.moonscript (lua52Packages.moonscript)</li>
    <li>luaPackages.mpack (lua52Packages.mpack)</li>
    <li>luaPackages.nvim-client (lua52Packages.nvim-client)</li>
    <li>luaPackages.penlight (lua52Packages.penlight)</li>
    <li>luaPackages.readline (lua52Packages.readline)</li>
    <li>luaPackages.say (lua52Packages.say)</li>
    <li>luaPackages.std__debug (lua52Packages.std__debug)</li>
    <li>luaPackages.std_normalize (lua52Packages.std_normalize)</li>
    <li>luaPackages.stdlib (lua52Packages.stdlib)</li>
    <li>luaPackages.vicious (lua52Packages.vicious)</li>
    <li>luaPackages.vstruct (lua52Packages.vstruct)</li>
    <li>luaPackages.wrapLua (lua52Packages.wrapLua)</li>
    <li>lua53Packages.alt-getopt</li>
    <li>lua53Packages.ansicolors</li>
    <li>lua53Packages.argparse</li>
    <li>lua53Packages.basexx</li>
    <li>lua53Packages.binaryheap</li>
    <li>lua53Packages.bit32</li>
    <li>lua53Packages.busted</li>
    <li>lua53Packages.cassowary</li>
    <li>lua53Packages.cjson</li>
    <li>lua53Packages.compat53</li>
    <li>lua53Packages.cosmo</li>
    <li>lua53Packages.coxpcall</li>
    <li>lua53Packages.cyrussasl</li>
    <li>lua53Packages.digestif</li>
    <li>lua53Packages.dkjson</li>
    <li>lua53Packages.fifo</li>
    <li>lua53Packages.inspect</li>
    <li>lua53Packages.ldbus</li>
    <li>lua53Packages.ldoc</li>
    <li>lua53Packages.lgi</li>
    <li>lua53Packages.linenoise</li>
    <li>lua53Packages.lpeg</li>
    <li>lua53Packages.lpeg_patterns</li>
    <li>lua53Packages.lpeglabel</li>
    <li>lua53Packages.lpty</li>
    <li>lua53Packages.lrexlib-gnu</li>
    <li>lua53Packages.lrexlib-pcre</li>
    <li>lua53Packages.lrexlib-posix</li>
    <li>lua5_3 (lua53Packages.lua)</li>
    <li>lua53Packages.lua-cmsgpack</li>
    <li>lua53Packages.lua-iconv</li>
    <li>lua53Packages.lua-lsp</li>
    <li>lua53Packages.lua-messagepack</li>
    <li>lua53Packages.lua-resty-http</li>
    <li>lua53Packages.lua-resty-jwt</li>
    <li>lua53Packages.lua-resty-openidc</li>
    <li>lua53Packages.lua-resty-openssl</li>
    <li>lua53Packages.lua-resty-session</li>
    <li>lua53Packages.lua-term</li>
    <li>lua53Packages.lua-toml</li>
    <li>lua53Packages.lua-yajl</li>
    <li>lua53Packages.lua-zlib</li>
    <li>lua53Packages.lua_cliargs</li>
    <li>lua53Packages.luacheck</li>
    <li>lua53Packages.luacov</li>
    <li>lua53Packages.luadbi</li>
    <li>lua53Packages.luadbi-mysql</li>
    <li>lua53Packages.luadbi-postgresql</li>
    <li>lua53Packages.luadbi-sqlite3</li>
    <li>lua53Packages.luadoc</li>
    <li>lua53Packages.luaepnf</li>
    <li>lua53Packages.luaevent</li>
    <li>lua53Packages.luaexpat</li>
    <li>lua53Packages.luaffi</li>
    <li>lua53Packages.luafilesystem</li>
    <li>lua53Packages.lualogging</li>
    <li>lua53Packages.luaossl</li>
    <li>lua53Packages.luaposix</li>
    <li>lua53Packages.luarepl</li>
    <li>lua53Packages.luarocks</li>
    <li>lua53Packages.luarocks-nix</li>
    <li>lua53Packages.luasec</li>
    <li>lua53Packages.luasocket</li>
    <li>lua53Packages.luasql-sqlite3</li>
    <li>lua53Packages.luassert</li>
    <li>lua53Packages.luasystem</li>
    <li>lua53Packages.luautf8</li>
    <li>lua53Packages.luazip</li>
    <li>lua53Packages.luuid</li>
    <li>lua53Packages.luv</li>
    <li>lua53Packages.luxio</li>
    <li>lua53Packages.lyaml</li>
    <li>lua53Packages.markdown</li>
    <li>lua53Packages.mediator_lua</li>
    <li>lua53Packages.moonscript</li>
    <li>lua53Packages.mpack</li>
    <li>lua53Packages.nvim-client</li>
    <li>lua53Packages.penlight</li>
    <li>lua53Packages.readline</li>
    <li>lua53Packages.say</li>
    <li>lua53Packages.std__debug</li>
    <li>lua53Packages.std_normalize</li>
    <li>lua53Packages.stdlib</li>
    <li>lua53Packages.vicious</li>
    <li>lua53Packages.wrapLua</li>
    <li>lua5_3_compat</li>
    <li>lua5_4</li>
    <li>lua5_4_compat</li>
    <li>mari0</li>
    <li>mpv</li>
    <li>mpv-unwrapped</li>
    <li>mrrescue</li>
    <li>nice-dcv-client</li>
    <li>nvimpager</li>
    <li>orthorobot</li>
    <li>pandoc</li>
    <li>perl530Packages.RPM2</li>
    <li>perl532Packages.RPM2</li>
    <li>pmix</li>
    <li>prosody</li>
    <li>python38Packages.osc</li>
    <li>rpm (python38Packages.rpm)</li>
    <li>ranger</li>
    <li>rdma-core</li>
    <li>sentry-cli</li>
    <li>sienna</li>
    <li>toluapp</li>
    <li>ucx</li>
    <li>unvanquished</li>
    <li>weechatScripts.weechat-matrix-bridge</li>
    <li>z-lua</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/interpreters/lua-5/interpreter.nix:121:3:
```
    |
121 |   meta = {
    |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

symphorien commented:
/rebase staging

symphorien commented on pkgs/development/interpreters/lua-5/interpreter.nix:
```diff
@@ -66,7 +72,8 @@ self = stdenv.mkDerivation rec {
     installFlagsArray=( TO_BIN="lua luac" INSTALL_DATA='cp -d' \
-      TO_LIB="${if stdenv.isDarwin then "liblua.${version}.dylib" else "liblua.a liblua.so liblua.so.${luaversion} liblua.so.${version}"}" )
+      TO_LIB="${if stdenv.isDarwin then "liblua.${version}.dylib"
+                else ("liblua.a" + lib.optionalString (!staticOnly) " liblua.so liblua.so.${luaversion} liblua.so.${version}" )}" )
```
that seems incorrect on darwin. Is it supported at all?

malbarbo replied:
This PR does not change the behavior on darwin, so I am assuming that it is correct.

malbarbo committed:
lua: remove unecessary postBuild for lua 5.3

I Think this is not necessary anymore since 34f8d7e15a3ca5762c84f867f70773674b002cb2

malbarbo commented on pkgs/development/interpreters/lua-5/default.nix:
```diff
@@ -19,10 +19,6 @@ rec {
 
     patches =
       lib.optionals stdenv.isDarwin [ ./5.2.darwin.patch ];
-
```
Fix `pkgsStatic.lua5_3` build.
